src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -471,6 +471,11 @@ subroutine atm_srk3(domain, dt, itimestep, exchange_halo_group)
          call mpas_pool_get_dimension(state, 'index_qg', index_qg)  
          call mpas_pool_get_dimension(state, 'index_nr', index_nr)
          call mpas_pool_get_dimension(state, 'index_ni', index_ni)
+         if(config_microp_scheme == 'mp_thompson_aerosols') then
```

Comment:
I don't think we need these conditional checks on `config_microp_scheme` here and elsewhere in `mpas_atm_time_integration.F`. If the aerosol-aware Thompson scheme isn't being used, these indices will be returned as -1 (the same as, e.g., `index_nr` and `index_ni`).

Reply:
you are right. I removed lines 474 and 478.